bundles/statistics/statsgrid/handler/SearchHandler.js

```diff
@@ -106,51 +106,64 @@ class SearchController extends StateHandler {
         this.updateState({
             loading: true
         });
-        this.service.getIndicatorList(this.state.selectedDatasource, (err, result) => {
-            const results = [];
-            if (err) {
-                // notify error!!
-                Oskari.log('Oskari.statistics.statsgrid.IndicatorSelection').warn('Error getting indicator list');
-                Messaging.error(this.loc('errors.indicatorListError'));
-                return;
-            }
-            const disabledIndicatorIDs = [];
-            result.indicators.forEach((ind) => {
-                const resultObj = {
-                    id: ind.id,
-                    title: Oskari.getLocalized(ind.name)
-                };
-                results.push(resultObj);
-                if (hasRegionSetRestriction) {
-                    const supportsRegionset = this.state.selectedRegionsets.some((iter) => {
-                        return ind.regionsets.indexOf(Number(iter)) !== -1;
+
+        const results = [];
+        const disabledIndicatorIDs = [];
+        const dataLoaded = new Promise((resolve, reject) => {
+            this.service.getIndicatorList(this.state.selectedDatasource, (err, result) => {
+                if (err) {
+                    // notify error!!
+                    Oskari.log('Oskari.statistics.statsgrid.IndicatorSelection').warn('Error getting indicator list');
+                    Messaging.error(this.loc('errors.indicatorListError'));
+                    reject(new Error(err));
+                    return;
+                }
+                const promises = result.indicators.map((ind) => {
+                    return new Promise((resolve, reject) => {
+                        const resultObj = {
+                            id: ind.id,
+                            title: Oskari.getLocalized(ind.name)
+                        };
+                        results.push(resultObj);
+                        if (hasRegionSetRestriction) {
+                            const supportsRegionset = this.state.selectedRegionsets.some((iter) => {
+                                return ind.regionsets.indexOf(Number(iter)) !== -1;
+                            });
+                            if (!supportsRegionset) {
+                                disabledIndicatorIDs.push(ind.id);
+                            }
+                        }
+                        resolve();
                     });
```

Comment:
There's no async operations here and the promises are always resolved. Using promise here just makes this more complex than it needs to be.